### DISCUSSION TITLE
Metamask account creation using social login

### DISCUSSION THREAD

**Discussion by subsixx** [Web3Auth/Web3Auth#586] [Help]
Hi guys,
I want to build a dapp where people can login via both social logins and metamask. I want to implement :- If user don't have any metamask account and login via google or any socials in my dapp, I want the web3auth SDK to create an metamask account for the user underhood so that the user can use this account to use my dapp. --- is this possible? ( I am a noob and  eager to make web3 application )

Thank you : )

**Comment by rinzler99** [comment]
Hey Anubhav,
Unfortunately, you cannot create metamask account with web3auth at the moment.

1.  While using Web3Auth, you have the flexibility of creating your own wallet within your application. So the user can export their private key and import into their metamask account.


2. To Implement login via social logins and metamask you can use web3auth sdk which will show web3auth modal for logging to your dapp. You can configure web3auth modal to use metamask and desired social logins [Reference](https://web3auth.io/docs/integration-builder?lang=react&chain=eth&customAuthentication=no&whitelabel=no&mfaParams=no&usingEmailPasswordless=no&rnWorkflowMode=expo&evmFramework=web3#step-3).

**Reply by subsixx** [reply]
Thank you
